### PR TITLE
PLT-7884 Remove all dependencies on plutus-apps repository

### PR DESCRIPTION
**Pull request by jhbertra**
- [x] Quarantined plutus-tx-plugin usage to `marlowe-plutus`
- [x] Refactored `marlowe-test`lib to just be reference and golden contracts, and Arbitrary instances.
- [x] Removed all `plutus-ledger` dependencies
- [x] Vendored `plutus-script-utils`
- [ ] Removed usage of typed validators

Pre-submit checklist:
- Branch
    - [ ] Tests are provided (if possible)
    - [ ] [Test report is updated](https://github.com/input-output-hk/marlowe-cardano/blob/main/marlowe/test/test-report.md) (if relevant)
    - [x] Commit sequence broadly makes sense
    - [x] Key commits have useful messages
    - [x] Relevant tickets are mentioned in commit messages
    - [ ] Formatting, PNG optimization, etc. are updated
- PR
    - [x] Self-reviewed the diff
    - [x] Useful pull request description
        - Review required
        - [x] Substantial changes to code, test, or documentation
        - [ ] Change made to Marlowe validator (@bwbush and @palas must be included as reviewers)
        - Review not required
        - [ ] Minor changes to non-critical code, documentation, nix derivations, configuration files, or scripts
        - [ ] Formatting, spelling, grammar, or reorganization
    - [x] Reviewer requested

### CODE REVIEW

File: stack.yaml
Comment (bwbush) [inline stack.yaml:1]: Thanks!

**Review by bwbush — COMMENTED** [review]
Looks good, but I need to understand why the script hashes changed before I approve this.

**Review by bwbush — COMMENTED** [review]
@jhbertra, I'm unsure of 96f49e74f750460a2f78f0b43806f3dc54ba2b9c. Should we expect changes in the golden tests for MarloweTxCommand because of the plutus-apps removal?

**Comment by bwbush** [comment]
:frowning: CI is failing due to lack of disk space. It succeeds locally.

**Comment by jhbertra** [comment]
> @jhbertra, I'm unsure of [96f49e7](https://github.com/input-output-hk/marlowe-cardano/commit/96f49e74f750460a2f78f0b43806f3dc54ba2b9c). Should we expect changes in the golden tests for MarloweTxCommand because of the plutus-apps removal?
No, these should have been updated in #694 - I'm not sure how those tests passed without the golden tests having been updated.

**Pull request merged**